rest/system_metrics_routes.go

```diff
@@ -0,0 +1,70 @@
+package rest
+
+import (
+	"context"
+	"net/http"
+	"strconv"
+
+	"github.com/evergreen-ci/cedar/model"
+	"github.com/evergreen-ci/cedar/rest/data"
+	"github.com/evergreen-ci/gimlet"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/message"
+	"github.com/pkg/errors"
+)
+
+///////////////////////////////////////////////////////////////////////////////
+//
+// GET /system_metrics/type/{task_id}/{type}
+
+type systemMetricsGetByTypeHandler struct {
+	metricType string
+	opts       model.SystemMetricsFindOptions
+	sc         data.Connector
+}
+
+func makeGetSystemMetricsByType(sc data.Connector) gimlet.RouteHandler {
+	return &systemMetricsGetByTypeHandler{
+		sc: sc,
+	}
+}
+
+// Factory returns a pointer to a new systemMetricsGetByTypeHandler.
+func (h *systemMetricsGetByTypeHandler) Factory() gimlet.RouteHandler {
+	return &systemMetricsGetByTypeHandler{
```

Comment:
Out of curiosity, it seems like a lot of the REST code does this, but why not just return `makeGetSystemMetricsByType(h.sc)`?

Reply:
Not actually sure, something I always copied and pasted lol